File: OlypicGamesC++/Game.cpp
#include "Game.h"

Game::~Game() {
	for (Sport* sport : sports) {
		delete sport;
	}

	for (pair<string, vector<Competitor*>> p : countries) {
		for (Competitor* comp : p.second) {
			delete comp;
		}
	}
}

Sport* Game::getSport(string& name) {
	//eather returns already created sport or return newly created
	for (Sport* sport : sports) {
		if (sport->getName() == name) return sport;
	}
	Sport* newSport = new Sport(name);
	sports.push_back(newSport);
	return newSport;
}
bool Game::hasSport(string& name) {
	for (Sport* sport : sports) {
		if (sport->getName() == name) return true;
	}
	return false;
}

void Game::addCompetitor(string& country, Competitor* competitor) {
	//add competitor
	countries[country].push_back(competitor);
}

File: OlympicGamesJava/logic/Game.java
package logic;

import java.util.Vector;
import java.util.HashMap;

public class Game {
	private int year;
	private String season, city;
	private Vector<Sport> sports = new Vector<Sport>();
	private HashMap<String, Vector<Competitor>> countries = new HashMap<String, Vector<Competitor>>();
	
	public Game(int y, String s, String c){
		this.year = y; this.season = s; this.city = c;
	}
	public int getYear() { return this.year; }
	public String getSeason() { return this.season; }
	public String getCity() { return this.city; }
	public HashMap<String, Vector<Competitor>> getCountries() { return countries; }

	public Sport getSport(String name) {
		for (Sport sport : sports) {
			if (sport.getName().equals(name)) return sport;
		}
		Sport newSport = new Sport(name);
		sports.add(newSport);
		return newSport;
	}
	public boolean hasSport(String name) {
		for (Sport sport : sports) {
			if (sport.getName().equals(name)) return true;
		}
		return false;
	}
	public void addCompetitor(String country, Competitor competitor) {
		if( countries.get(country) == null )
			countries.put(country, new Vector<Competitor>());
		countries.get(country).add(competitor);
	}
}
